daemon/cmd/config.go

```diff
@@ -135,6 +140,39 @@ func patchConfigHandler(d *Daemon, params PatchConfigParams) middleware.Responde
 	return api.Error(PatchConfigFailureCode, msg)
 }
 
+// getIPLocalReservedPorts returns a comma-separated list of ports which
+// we need to reserve in the container network namespace.
+// These ports are typically used in the host network namespace and thus can
+// conflict when running with DNS transparent proxy mode.
+// This is a workaround for cilium/cilium#31535
+func getIPLocalReservedPorts(d *Daemon) string {
+	if option.Config.ContainerIPLocalReservedPorts != defaults.ContainerIPLocalReservedPortsAuto {
+		return option.Config.ContainerIPLocalReservedPorts
+	}
+
+	if !option.Config.DNSProxyEnableTransparentMode {
+		return "" // no ports to reserve
+	}
+
+	// Reserves the WireGuard port. This is usually part of the ephemeral port
+	// range and thus may conflict with the ephemeral source port of DNS clients
+	// in the container network namespace.
+	var ports []string
+	if option.Config.EnableWireguard {
+		ports = append(ports, strconv.Itoa(wgTypes.ListenPort))
+	}
+
+	// Reserves the VXLAN port. This is not part of the ephemeral port range by
+	// default, but is user configurable and thus should be included regardless.
+	// The Linux kernel documentation explicitly allows to reserve ports which
+	// are not part of the ephemeral port range, in which case this is a no-op.
+	if d.tunnelConfig.Protocol() == tunnel.VXLAN {
+		ports = append(ports, fmt.Sprintf("%d", d.tunnelConfig.Port()))
+	}
+
```

Comment:
Shouldn't we also consider `tunnel.Geneve` here?

Reply:
Yes. I forgot we still support it

Reply:
https://github.com/cilium/cilium/pull/32421